client/packages/programs/src/JsonForms/components/Search/usePatientSearchQuery.tsx

```diff
@@ -0,0 +1,22 @@
+import { FilterBy } from '@openmsupply-client/common';
+import { Document } from '@openmsupply-client/system/src/Patient/api/hooks/document';
+
+export const usePatientSearchQuery = (searchFilter: FilterBy | undefined) => {
+  const { data, error, isLoading } = Document.usePatientFullSearch({
+    first: 10,
+    offset: 0,
+    sortBy: { key: 'lastName', direction: 'asc' },
+    filterBy: searchFilter,
+  });
+
+  const results =
+    // If patient has a full document field, use that since it'll make more data
+    // available. Otherwise just use the basic Patient fields
+    data?.nodes.map(patient => patient.document?.data ?? patient) ?? [];
+
+  return {
```

Comment:
There might actually be existing patients without document. And we want to include these in the search results. Could you please change it so that it works for the `patient` object instead for the `patient.document.data`? If you want to extract custom data from the document you could still specify this explicitly in the saveFields options...

Reply:
This will include all patients in the search results. It just preferentially adds the full document to the result, but if there isn't one it'll add the base patient data.

I don't see a downside to doing it this way, it'll work for patients with or without documents. (If the "save fields" don't exist on the resulting patient, they just get silently ignored (in `handlePatientSelect`))

Reply:
Problem is, its very hard to verify that these types are compatible. I just tried it out and (created a PR into this branch) and there were indeed only smaller unrelated conflicts. However, with the types in place the compiler can help now and its hopefully clearer what you get back from the hook.

Reply:
Okay, so I can explicitly define the results to be `PatientRowFragment[]`: 8ebc95999

That ensures the type in the component for `patient` is correct. (The "document" type would only ever be a superset of "patient", so shouldn't cause problems)

Reply:
Could you please have a look at the PR I created? https://github.com/openmsupply/open-msupply/pull/2319

The problem is that you assume the data is a PatientRowFragment which is not completely true (and hard to verify). What we know is that the patient data follows the Patient schema, so I added the schema patient type and then cast the  `data: any` type to this schema type. From there I let the compiler figure out if everything fits together.

Reply:
Okay, I'll merge this now and then we can address any remaining concerns with this in your PR. Thanks.